### PR TITLE
drivers: dac: improve test coverage

### PR DESCRIPTION
**Pull request by martinjaeger**
This PR contains a few commits to get full test coverage of all DAC drivers and all boards tagged with `dac` support.

Unfortunately, not all drivers were covered by the build-tests before.

Only the `dac_sam` driver failed to build because `zephyr/kernel.h` was not included. This was fixed in a dedicated commit.

See individual commit messages for further details.

### CODE REVIEW

**Review by nandojve — DISMISSED** [review]
LGTM!

File: drivers/dac/dac_sam.c [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #include <zephyr/drivers/dac.h>
 #include <zephyr/drivers/pinctrl.h>
 #include <zephyr/drivers/clock_control/atmel_sam_pmc.h>
-
+#include <zephyr/kernel.h>
```
Comment (henrikbrixandersen) [inline drivers/dac/dac_sam.c:23]: This looks like a bug that ought to be handled in a separate PR and backported to v3.3-branch?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: drivers/dac/dac_sam.c (hunk 1 above)
Comment (martinjaeger) [reply drivers/dac/dac_sam.c:23]: Yeah, you are probably right ;)

Done: #56357

Will rebase after that one is merged.

**Comment by martinjaeger** [comment]
Rebased after merge of #56357.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by nandojve — APPROVED** [review]
